src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java

```diff
@@ -259,28 +288,96 @@ private VersionNumberBuildInfo incBuild(AbstractBuild build, PrintStream log) th
             // increment total builds
             buildsAllTime = info.getBuildsAllTime() + buildInc;
         }
+        
         // have we overridden any of the version number info?  If so, set it up here
         boolean saveOverrides = false;
-        if (this.oBuildsToday >= 0) {
-            buildsToday = oBuildsToday;
-            oBuildsToday = -1;
-            saveOverrides = true;
+        Pattern pattern = Pattern.compile(envVarPattern);
```

Comment:
As the pattern `envVarPattern` is actually a constant variable (at least that's what I assumed by making it `private static final`) and it was already tested by me and compiled without an exception, this should never throw an exception.
And if it does, that means that a plugin-developer modified the constant and did something wrong. In that case, it probably is better to let the exception escape this method so that the plugin-developer notices it.

Therefore, I would prefer to leave this as is. Is this fine with you?